cou/steps/plan.py

```diff
@@ -52,24 +54,162 @@
 logger = logging.getLogger(__name__)
 
 
-async def generate_plan(analysis_result: Analysis, backup_database: bool) -> UpgradePlan:
+def pre_plan_sane_checks(args: Namespace, analysis_result: Analysis) -> None:
+    """Pre checks to generate the upgrade plan.
+
+    :param args: CLI arguments
+    :type args: Namespace
+    :param analysis_result: Analysis result.
+    :type analysis_result: Analysis
+    """
+    is_valid_openstack_cloud(analysis_result)
+    is_supported_series(analysis_result)
+    is_highest_release_achieved(analysis_result)
+
+    if args.upgrade_group == DATA_PLANE:
+        is_data_plane_ready_to_upgrade(analysis_result)
+
+
+def is_valid_openstack_cloud(analysis_result: Analysis) -> None:
+    """Check if the model passed is a valid OpenStack cloud.
+
+    :param analysis_result: Analysis result
+    :type analysis_result: Analysis
+    :raises NoTargetError: When cannot determine the current OS release
+        or Ubuntu series.
+    """
+    if not analysis_result.current_cloud_os_release:
+        raise NoTargetError(
+            "Cannot determine the current OS release in the cloud. "
+            "Is this a valid OpenStack cloud?"
+        )
+
+    if not analysis_result.current_cloud_series:
+        raise NoTargetError(
+            "Cannot determine the current Ubuntu series in the cloud. "
+            "Is this a valid OpenStack cloud?"
+        )
+
+
+def is_supported_series(analysis_result: Analysis) -> None:
+    """Check the Ubuntu series of the cloud to see if it is supported.
+
+    :param analysis_result: Analysis result.
+    :type analysis_result: Analysis
+    :raises OutOfSupportRange: When series is not supported.
+    """
+    supporting_lts_series = ", ".join(LTS_TO_OS_RELEASE)
+    # series already checked at is_valid_openstack_cloud
+    if (
+        current_series := analysis_result.current_cloud_series
+    ) and current_series not in LTS_TO_OS_RELEASE:
```

Comment:
This should work fine and it is more readable. 
```suggestion
    if (
        current_series := analysis_result.current_cloud_series
        and current_series not in LTS_TO_OS_RELEASE
    ):
```

Reply:
I've tried before and It does not work.[0](https://peps.python.org/pep-0572/#exceptional-cases)

Reply:
Ok, but this should:
```python
if (
        (current_series := analysis_result.current_cloud_series)
        and current_series not in LTS_TO_OS_RELEASE
):
```
However, IMO this does not bring any benefit than using:
```python
current_series = analysis_result.current_cloud_series
if current_series not in LTS_TO_OS_RELEASE:
```
Since, we will define that variable in any case, and it will not happen only in some cases.